TTML1 3ED tests

palemieux opened the pull request:
Provides tests for all testable changes from TTML1 2ED to TTML1 3ED, as follows:

- clarified default value blur radius in tts:textOutline (#351):

TextOutline.ttml

- update tts:lineHeight='normal' algorithm to match TTML2 (#355):

LineHeight.ttml

- clarified tts:fontSize semantics (#206, #214 and #215):

FontSizeEmInherited.ttml
TwoValueRelativeFontSize.ttml
FontSizeEmRegion.ttml

- specified timing semantics for `br` elements (#339)

BrImplicitDuration.ttml

- clarified tts:direction on `p` elements (#324)

Direction.ttml

- clarified LWSP in color expressions (#322)

Color.html

- clarified validation of undefined items in TT namespaces (#251)

UnknownTTAtribute.ttml

- clarified anonymous span construction, implicit duration for anonymous spans, implicit duration for spans with text nodes exclusively, and implicit duration for `set` (#310, #193)

AnonymousSpanImplicitDuration.ttml
PlainSpanImplicitDuration.ttml
SetImplicitDuration.ttml
  
- clarified that LWSPs are permitted in tts:fontFamily (#248)

FontFamily.ttml

- added initial value fallback to non-inheritable properties (#200)

NonInheritedInitial.ttml

- added `br` to elements to which style applies (#249)

Br.ttml

- specified application defaults for frame rate multiplier and sub-framerate (#192)

_not testable_

- prevent pruning of `br` elements (#197)

Br.ttml

- handled forward interoperability of attribute extensions in TT namespaces (#196)

ForwardInterop.ttml

- prevent <set> elements from being pruned in ISD construction (#216)

Animation.ttml

- clarified that the inherited value of tts:lineHeight='normal' is 'normal' (#220)

LineHeight.ttml

- clarified application of tts:visibility to child elements (#207)

Visibility.ttml

- clarified tts:textDecoration inheritance (#219)

TextDecoration.ttml

- clarified cell metrics for tts:textOutline (#218)

TextOutline.ttml

- clarified tts:zIndex stacking context (#241)

ZIndex.ttml

palemieux committed:
Initial commit of TTML1 3ED tests

palemieux committed:
Merge remote-tracking branch 'origin/master' into feature/3ED-tests

nigelmegitt commented on testsuite/Third Edition/AnonymousSpanImplicitDuration.ttml:
```diff
@@ -0,0 +1,20 @@
+  <body>
+    <div>
+      <p begin="0s" end="1s" timeContainer="seq" tts:backgroundColor="blue">
+        No text should be displayed.
```
Fine point: maybe change to "This text should not be displayed" so if it _is_ displayed that's clearly an error.

nigelmegitt commented on testsuite/Third Edition/BrImplicitDuration.ttml:
```diff
@@ -0,0 +1,18 @@
+  <body>
+    <div>
+      <p begin="0s" end="1s" timeContainer="seq" tts:backgroundColor="blue">
+        No text or linebreak<br/>should be displayed.
```
If text is not displayed but a line break is displayed, presumably that would move the text in the second paragraph down a line. We could highlight that by changing the text of the second paragraph below to say "this text should be displayed at the top" before the break.

nigelmegitt replied:
During call, @palemieux pointed out there would be a background colour, and I suggested making it a different colour to the following p.

palemieux replied:
Change color of the second `p`.

nigelmegitt commented on testsuite/Third Edition/Color.ttml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--Modifications Copyright (c) 2017, Pierre-Anthony Lemieux (pal@palemieux.com)-->
```
Really? I don't think I've seen a copyright notice on a test suite test before.

palemieux replied:
Leftover from a long time ago. Fixed at https://github.com/w3c/ttml1/pull/361/commits/32d0ae13c0e0f668634a457f9948651a46dde6a4

nigelmegitt commented on testsuite/Third Edition/Direction.ttml:
```diff
@@ -0,0 +1,17 @@
+  </head>
+  <body>
+    <div>
+      <p tts:direction="rtl">
```
Would be worth some kind of note that this only defines weak directionality so it shouldn't be ltr.

nigelmegitt replied:
Addressed in b1753b0

nigelmegitt commented on testsuite/Third Edition/Direction.ttml:
```diff
@@ -0,0 +1,17 @@
+    <div>
+      <p tts:direction="rtl">
+        Left to right<br/>
+        משמאל לימין<br/>
```
This text means "left to right" in English.

palemieux replied:
מימין לשמאל works better?

nigelmegitt replied:
Confirmed with a native Hebrew speaker, that's good.

nigelmegitt replied:
It means "From right to left".

skynavga reviewed (commented):
Regarding line height tests, the new text that drives this test, "When a single relative `<length>` ...", was previously implied in TTML1 2e, as evidenced by the second sentence in the fourth comment in 8.2.9 of 2e.

palemieux committed:
Remove copyright statements

nigelmegitt reviewed (commented):
Partial review due to lack of time.

nigelmegitt commented on testsuite/Third Edition/Br.ttml:
```diff
@@ -0,0 +1,13 @@
+      <p begin="0s" end="10s">This text must be on the first line.<br tts:display="none"/>This text on a second line.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL

nigelmegitt commented on testsuite/Third Edition/BrImplicitDuration.ttml:
```diff
@@ -0,0 +1,18 @@
+      <p begin="0s" end="1s" timeContainer="seq" tts:backgroundColor="blue">
+        No text or linebreak<br/>should be displayed.
+      </p>
+      <p begin="0s" end="1s" tts:backgroundColor="blue">
```
Change to a different background color to distinguish from first paragraph please, as discussed.

nigelmegitt commented on testsuite/Third Edition/Color.ttml:
```diff
@@ -0,0 +1,14 @@
+      <p tts:color="rgb( 0,   128, 0 )" begin="0s" end="10s">This text must be green.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/FontFamily.xml:
```diff
@@ -0,0 +1,16 @@
+  </head>
+  <body>
+    <div>
+      <p begin="0s" end="10s">The last words must be <span tts:fontFamily='InexistantFont, proportionalSerif'>using a proportional serif font</span>.</p>
```
Unusual use of single quotes here, why not use double quotes as per other attributes and examples?

nigelmegitt commented on testsuite/Third Edition/FontSizeEmInherited.ttml:
```diff
@@ -0,0 +1,18 @@
+      <span tts:fontSize="2em">Double font size</span></p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/FontSizeEmRegion.ttml:
```diff
@@ -0,0 +1,16 @@
+      <p>Font size is 1/5 the height of the root container</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/ForwardInterop.ttml:
```diff
@@ -0,0 +1,13 @@
+      <p>This text must not be half the height of the root container.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/NonInheritedInitial.ttml:
```diff
@@ -0,0 +1,17 @@
+      <p begin="0s" end="10s" region="r1">This text is aligned with<br/>the before edge of the region.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/PlainSpanImplicitDuration.ttml:
```diff
@@ -0,0 +1,20 @@
+  <body>
+    <div>
+      <p begin="0s" end="1s" timeContainer="seq" tts:backgroundColor="blue">
+        <span>No text should be displayed.</span>
```
Suggest changing text to "This text should not be displayed" - it is not true that no text should be displayed since the text from line 16 should appear.

nigelmegitt commented on testsuite/Third Edition/PlainSpanImplicitDuration.ttml:
```diff
@@ -0,0 +1,20 @@
+    </div>
+    <div>
+      <p begin="0s" end="1s" tts:backgroundColor="blue">
+        <span>Text should displayed.</span>
```
As per above comment, suggest changing to "This text should be displayed.".

nigelmegitt commented on testsuite/Third Edition/PlainSpanImplicitDuration.ttml:
```diff
@@ -0,0 +1,20 @@
+      </p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/TextDecoration.ttml:
```diff
@@ -0,0 +1,17 @@
+      <p begin="0s" end="10s">The last two words in this caption are<span tts:textDecoration="noUnderline"> <br/>not underlined.</span></p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/SetImplicitDuration.ttml:
```diff
@@ -0,0 +1,22 @@
+      </p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/TextOutline.ttml:
```diff
@@ -0,0 +1,17 @@
+      <p begin="0s" end="10s" style="s1">This text has a red outline with a radius equal to 10% of the font size.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/TwoValueRelativeFontSize.ttml:
```diff
@@ -0,0 +1,21 @@
+      </p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/UnknownTTAtribute.ttml:
```diff
@@ -0,0 +1,13 @@
+      <p tts:foo="7.5c">This text must appear.</p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/Visibility.ttml:
```diff
@@ -0,0 +1,17 @@
+    	<p begin="0s" end="10s"><span tts:visibility="visible">All the words in this caption are visible.</span></p>
+    </div>
+  </body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/ZIndex.ttml:
```diff
@@ -0,0 +1,32 @@
+			<p region="r6" begin="0s" end="1s">tts:zIndex="-1000"</p>
+		</div>
+	</body>
+</tt:tt>
```
Missing EOL.

nigelmegitt commented on testsuite/Third Edition/ZIndex.ttml:
```diff
@@ -0,0 +1,32 @@
+		</layout>
+	</head>
+	<body>
+		<div timeContainer="seq">
```
Using `seq` means that no overlapping regions are created during this test, so no z-order comparisons can be done. Is that correct? Is this test only checking that implementations handle the value space of the `tts:zIndex` style processing?

nigelmegitt replied:
Discussed in yesterday's TTWG call; the intent here is only to verify that all the values of `tts:zIndex` are supported. They are all valid.

nigelmegitt reviewed (changes requested):
Some changes requested, mainly these tests look good.

palemieux committed:
Address review comments

nigelmegitt committed:
Resolve comments as discussed in TTWG meeting 2018-09-07

Add or improve descriptive metadata and in the case of `PlainSpanImplicitDuration.ttml`, clarify the text that should (or should not) appear.

nigelmegitt commented:
@palemieux please check you are happy with the changes made in b1753b0 as per the TTWG meeting yesterday.

palemieux commented:
@skynavga Are you arguing at https://github.com/w3c/ttml1/pull/361#pullrequestreview-144873178 that `testsuite/Third Edition/TwoValueRelativeFontSize.ttml` is not needed?

nigelmegitt commented:
Reviewing https://github.com/w3c/ttml1/pull/361#pullrequestreview-144873178 and comparing 2e with 3e, the substantive change is essentially a movement of text from an informative note to a normative section. Have we got any evidence that the clarified behaviour is what was already implemented? I believe that is what the Director is seeking.

The alternative would be that some implementation supports 2 value font sizes but uses a different basis for calculating the font size dimensions, for example always using the height of the cell as the basis even for the horizontal component.

palemieux committed:
Merge branch 'master' into feature/3ED-tests

nigelmegitt commented:
Travis build stalled on this, I've restarted it.

nigelmegitt commented:
Build succeeded.

palemieux committed:
Removed IMSC profile signaling
Changed xml extenstion to ttml

palemieux committed:
Clarify tts:fontSize tests

palemieux commented:
> Have we got any evidence that the clarified behaviour is what was already implemented?

@skynavga Can you confirm that 3ED captures current TTPE behavior?

palemieux committed:
Merge branch 'master' into feature/3ED-tests
